File: 1주차/beomjookim/reverse_vowels_of_a_string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        # 한번의 iteration 안에 끝내는 좋은 방법이 있었다...
        # reverse니까 two pointer 활용하면 됨.
        # 그리고 s를 그대로 두고 진행하면 너무 느리니까, list로 바꿨다가 다시 바꿔서 리턴하기.

        vowels = "aeiouAEIOU"

        temp_s = list(s)

        i, j = 0, len(s)-1

        while True:
            while s[i] not in vowels and i < j: i += 1
            while s[j] not in vowels and j > i: j -= 1

            if i >= j: return "".join(temp_s)
            else:
                temp_s[i], temp_s[j] = temp_s[j], temp_s[i]
                i, j = i+1, j-1

File: 1주차/mijip0320/reverse_vowels_of_a_string.js
var reverseVowels = function(s) {
   const vowels = 'aAeEiIoOuU';
    let vow = ""; //vowels that are included in s
    let i = 0;
    let revStr = '';

    for (let c of s) {
      if(vowels.includes(c)){
            vow += c;
        }
    }

    //reverse
    vow = vow.split('').reverse().join('')
    for (let char of s) {
        if (vowels.includes(char)) {
            revStr += vow[i] //add vowel to revStr
            i++
        } else {
            revStr += char //add letter to revStr
        }
    }
    
    return revStr;
};